src/pydocstyle/config.py

```diff
@@ -208,7 +208,10 @@ def _get_config(self, node):
         if path in self._cache:
             return self._cache[path]
 
-        config_file = self._get_config_file_in_folder(path)
+        if self._options.config_file:
```

Comment:
I don't think this should affect the config files search logic, only the initial directory/file the search is started from. Meaning, `--config=PATH`, where `PATH` can be a standard config file path or a directory path (so you could check if your file complies with standards of files in a directory, no matter where their configuration is in the directory tree).